test/Plutip/Contract.purs

```diff
@@ -784,6 +795,80 @@ suite = do
                 ]
             )
 
+    test "GetScriptByHash" do
+      let
+        distribution :: InitialUTxOs
+        distribution = [ BigInt.fromInt 2_000_000_000 ]
+
+      withWallets distribution \alice -> do
+        withKeyWallet alice do
+          validator1 <- AlwaysSucceeds.alwaysSucceedsScript
+          validator2 <- alwaysSucceedsScriptV2
+          let
+            validatorRef1 = PlutusScriptRef $ unwrap validator1
+            validatorRef2 = PlutusScriptRef $ unwrap validator2
+            useScriptAndGetByHash validator vhash = do
+              txId <- AlwaysSucceeds.payToAlwaysSucceeds vhash
+              awaitTxConfirmed txId
+              -- For Kupo (used inside) to see script, its utxo must be spent
```

Comment:
This isn't entirely true, Kupo will also index script references in outputs. For eg `mustPayToScriptWithScriptRef` is enough to make kupo aware in 1 transaction. That being said, what we have now is probably easier

Reply:
Thanks. I did not found mention of what you say in Kupo API docs. It says:

> However, a periodic garbage collector will automatically remove scripts that aren't associated to any matched output

https://cardanosolutions.github.io/kupo/#tag/Scripts

Reply:
> By default, the engine indexes every script present in transaction witnesses and inline scripts

Only a transaction which spends a script output will include the script in the witness set, but we also have inline scripts in utxos which will get indexed by Kupo.